builtin/providers/aws/resource_aws_db_event_subscription.go

```diff
@@ -0,0 +1,326 @@
+			log.Printf("[WARN] No RDS Event Subscription by name (%s) found", name)
+			d.SetId("")
+			return nil, nil
+		}
+		return nil, fmt.Errorf("Error reading RDS Event Subscription %s: %s", name, err)
+	}
+
+	if len(describeResp.EventSubscriptionsList) != 1 {
+		return nil, fmt.Errorf("Unable to find RDS Event Subscription: %#v", describeResp.EventSubscriptionsList)
+	}
+
+	return describeResp.EventSubscriptionsList[0], nil
+}
+
+func resourceAwsDbEventSubscriptionUpdate(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+
+	d.Partial(true)
+
+	// check/update tags, missing from go-aws-sdk?
+	if d.HasChange("enabled") || d.HasChange("event_categories") ||
+		d.HasChange("sns_topic") || d.HasChange("source_type") {
+
+		eventCategoriesSet := d.Get("event_categories").(*schema.Set)
+		eventCategories := make([]*string, eventCategoriesSet.Len())
+		for _, eventCategory := range eventCategoriesSet.List() {
+			eventCategories = append(eventCategories, aws.String(eventCategory.(string)))
+		}
+
+		_, err := rdsconn.ModifyEventSubscription(&rds.ModifyEventSubscriptionInput{
+			SubscriptionName: aws.String(d.Id()),
+			Enabled:          aws.Bool(d.Get("enabled").(bool)),
+			SnsTopicArn:      aws.String(d.Get("sns_topic").(string)),
+			SourceType:       aws.String(d.Get("source_type").(string)),
+			EventCategories:  eventCategories,
+		})
+
+		if err != nil {
+			return fmt.Errorf("Modifying RDS Event Subscription %s failed: %s", d.Id(), err)
+		}
+
+		log.Println(
+			"[INFO] Waiting for RDS Event Subscription modification to finish")
+
+		stateConf := &resource.StateChangeConf{
+			Pending:    []string{"modifying"},
+			Target:     []string{"active"},
+			Refresh:    resourceAwsDbEventSubscriptionRefreshFunc(d, meta.(*AWSClient).rdsconn),
+			Timeout:    40 * time.Minute,
+			MinTimeout: 10 * time.Second,
+			Delay:      30 * time.Second, // Wait 30 secs before starting
+		}
+
+		// Wait, catching any errors
+		_, err = stateConf.WaitForState()
+		if err != nil {
+			return fmt.Errorf("Modifying RDS Event Subscription %s failed: %s", d.Id(), err)
+		}
+	}
+
+	if arn, err := buildRDSEventSubscriptionARN(d, meta); err == nil {
+		if err := setTagsRDS(rdsconn, d, arn); err != nil {
+			return err
+		} else {
+			d.SetPartial("tags")
+		}
+	}
+	d.Partial(false)
+
+	return nil
+}
+
+func resourceAwsDbEventSubscriptionDelete(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+	deleteOpts := rds.DeleteEventSubscriptionInput{
+		SubscriptionName: aws.String(d.Id()),
+	}
+
+	if _, err := rdsconn.DeleteEventSubscription(&deleteOpts); err != nil {
+		rdserr, ok := err.(awserr.Error)
+		if !ok {
+			return fmt.Errorf("Error deleting RDS Event Subscription %s: %s", d.Id(), err)
+		}
+
+		if rdserr.Code() != "DBEventSubscriptionNotFoundFault" {
+			return fmt.Errorf("Error deleting RDS Event Subscription %s: %s", d.Id(), err)
+		}
+	}
```

Comment:
It's probably ok to have the check for resource which was deleted outside of Terraform in two places (Read + Delete), although in vast majority of other resources we only catch this in `Read`. In either case it should be accompanied w/ a WARN log message as it's a non-standard situation.